Release 153

gigxz opened the pull request:
https://github.com/greenriver/hmis-warehouse/pull/5141

gigxz committed:
Force reload on login (#1060)

* Force reload on login

* Fix comment

martha committed:
Add CI step for storybook test runner (#935)

* Add a11y addon to storybook

* Run yarn-deduplicate

* Add storybook test runner and CI step

* Run yarn-deduplicate again

* Add chromatic url (just fiddling)

* Still noodling around

* Add back a11y tests, ci should now fail

* Does chromatic step export a target url?

* oops, fix

* Add recipe - should fail since deps not installed

* Install missing packages

* Make storybook test a sub-step of build&test

* Run yarn and yarn-deduplicate after merge

* Maybe fix

* Maybe fix

* Add comments

* try running chromatic tests on ci

* run yarn-deduplicate

* Try running storybook tests against chromatic

* Try running storybook against chromatic

* Maybe fix

* Remove unneeded deps

* Skip a11y tests for now

* Uncomment tests to run

* Skip and fix failing tests

* README, yarn, and disable some tests

* Try increasing test timeout

* Clarify todos

gigxz committed:
Merge branch 'release-152' into release-153

ZachCMP committed:
React-hook-forms refactor of DynamicForm (#612)

Major refactoring of the dynamic form system to use react-hook-form
* Replace custom form state management with react-hook-form
* Add new hooks for field status, autofill, and value watching
* Split DynamicForm into smaller, focused components
* Improve performance by isolating re-renders
* preload picklist values before displaying form
* Attempt to keep backwards compatibility with existing form behavior
* minor upgrade to react-hook-form
* fixes issue with dynamic bound constraints on numeric inputs

No changes to form definition schema or submission format. Should improve form performance, especially for large forms

---------

Co-authored-by: Gig Ashton <gig@greenriver.org>
Co-authored-by: Theron Toomey <ttoomey@greenriver.org>

gigxz committed:
Edit records using original form (#1063)

gigxz committed:
Merge branch 'release-152' into release-153

gigxz committed:
Update HMIS color palette and Button styles, remove secondary color usage (#1007)

gigxz committed:
Fix color bug on disabled ButtonLink (#1071)

gigxz committed:
Updated enrollment workflow (#1065)

gigxz committed:
Allow nav item to specify active paths (#1072)

gigxz committed:
Fix broken input bg from theme pr (#1074)

gigxz committed:
Fix visual form bug (#1073)

* Add back support for form variant for modals

* Dont render empty space for errors

gigxz committed:
Make collapsible card accessible (#1075)

gigxz committed:
Fix back to enrollment path from assessment, fix protected IDs bug (#1077)

* Make collapsible card accessible

* ContentId always mounted

* Fix back to enrollment path from assessment

* Fix for individual assmt

* Handle bad householdId param

* Handle less gracefully

* Dont hash householdId in URL

* Address feedback

* Move loading to top level

gigxz committed:
Fix loading flicker on service form (#1079)

* Fix loading flicker on service form

* Rename var

gigxz committed:
Fix assessment print bug, remove adjustValue (#1080)

gigxz commented:
@martha any idea why the builds are failing on storybook? I'm having a hard time parsing it, maybe chromatic is down or something. Storybook is working for me locally.

gigxz committed:
Bug fix for multi-choice fields with remote picklist (#1081)

* Bug fix for multi-choice fields with remote picklist

* Skip empty array

martha commented:
@gigxz not sure, but I wonder if it was a rate limiting thing? Now that we have set chromatic build to happen on every PR we might be hitting a quota. But it looks like it did succeed this morning.

gigxz committed:
Address QA feedback on color palette (#1082)

* Remove links color

* Address feedback on side nav

* visibility toggle

* Revert padding

gigxz committed:
Upgrade dompurify (#1083)

gigxz committed:
More followups on link color (#1084)

ttoomey committed:
6043-rhf-bugs (#1091)

ttoomey committed:
6043 rhf fix (#1092)

Use rhf dirtyFields instead of isDirty to avoid false-positives (dirty
forms block tab nav in household assessment)

gigxz committed:
Fix initialSelected option list behavior

gigxz committed:
Dont set initialValue for read only mode

gigxz committed:
Remove early return

gigxz committed:
Merge pull request #1094 from greenriver/gig/patch-initial-selected-option-list

Fix initialSelected option list behavior

gigxz committed:
Merge branch 'release-152' into release-153

gigxz committed:
Merge branch 'stable' into release-153
